.github/workflows/dependency-check.yml

```
@@ -0,0 +1,49 @@
+name: Dependency Check
+
+on:
+  pull_request:
+    paths:
+      - 'go.mod'
+      - 'go.sum'
+      - '.github/workflows/dependency-check.yml'
+
+jobs:
+  dependency-check:
+    runs-on: ubuntu-latest
+    name: Dependency Check
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+      - id: all
+        run: |
+          echo "dependencies<<EOF" >> $GITHUB_OUTPUT
+          # Extract all dependencies from go.mod (include indirect dependencies and comments)
+          sed -n '/require (/,/)/{/require (/!{/)/!p;};}' go.mod | sed 's/^[[:space:]]*//' | tee -a $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+      - id: unreleased
+        env:
+          DEPENDENCIES: ${{ steps.all.outputs.dependencies }}
+        run: |
+          echo "dependencies<<EOF" >> $GITHUB_OUTPUT
+          # Find all dependencies that use prerelease versions (exclude v0.0.0 versions)
+          grep -Pv 'v0\.0\.0|v[0-9]+\.[0-9]+\.[0-9]+([^-]|$)' <<< "$DEPENDENCIES" | tee -a $GITHUB_OUTPUT
```

first match in this seems redundant, no? `v0.0.0` is matched by the second part, or am I missing something?

From what I can tell, `[^-]` is making allowance for the two  `+incompatible` packages, is that correct? Or is there something more broad? Having an open ended "if it doesn't have a `-` then it can have anything you like" is probably not ideal here.

`+incompatible` is a [Golang-specific thing](https://go-review.googlesource.com/c/go/+/124384) and I think we could just accommodate that directly. So maybe it becomes something like:

```suggestion
          grep -Pv 'v[0-9]+\.[0-9]+\.[0-9]+(\+incompatible)?$' <<< "$DEPENDENCIES" | tee -a $GITHUB_OUTPUT
```

---

oh, nmind re `v0.0.0`, it's open-ended on the back so it's for the non-tagged packages, we can be more specific I think because `v0.0.0` can also be a valid version and you could have `-whatever` on the back of it.

```suggestion
          grep -Pv 'v0\.0\.0-[0-9]{14}-[0-9a-f]{7,}$|v[0-9]+\.[0-9]+\.[0-9]+(\+incompatible)?$' <<< "$DEPENDENCIES" | tee -a $GITHUB_OUTPUT
```